packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -308,17 +334,23 @@ export function useGridColumns(
         return params;
       }
 
-      const columnVisibilityModel = context.stateToRestore.columns?.columnVisibilityModel;
-      if (columnVisibilityModel != null) {
+      const columnVisibilityModelToImport = context.stateToRestore.columns?.columnVisibilityModel;
+      const initialState = context.stateToRestore.columns;
+      if (columnVisibilityModelToImport != null || initialState != null) {
         const columnsState = createColumnsState({
```

Comment:
Perhaps
Exporting everything was simpler to avoid this can of "problems".
But if we want to only export what's interesting.
That means storing an additional `initialDimensions` key inside the columns state to be able to access this data right ?